File: fsharp/space-age/SpaceAgeTest.fs
﻿module SpaceAgeTests

open NUnit.Framework
open SpaceAge

[<TestFixture>]
type SpaceAgeTest() =
    
    [<Test>]
    member tests.Age_in_seconds() =
        Assert.That<decimal>(SpaceAge(1000000m).Seconds, Is.EqualTo(1000000))

    [<Test>]
    member tests.Age_on_earth() =
        Assert.That<decimal>(SpaceAge(1000000000m).onEarth, Is.EqualTo(31.69m))

    [<Test>]
    member tests.Age_on_mercury() =
        let spaceAge = SpaceAge(2134835688m)

        Assert.That<decimal>(spaceAge.onEarth, Is.EqualTo(67.65m))
        Assert.That<decimal>(spaceAge.onMercury, Is.EqualTo(280.88m))

    [<Test>]
    member tests.Age_on_venus() =
        let spaceAge = SpaceAge(189839836m)

        Assert.That<decimal>(spaceAge.onEarth, Is.EqualTo(6.02m))
        Assert.That<decimal>(spaceAge.onVenus, Is.EqualTo(9.78m))

    [<Test>]
    member tests.Age_on_mars() =
        let spaceAge = SpaceAge(2329871239m)

        Assert.That<decimal>(spaceAge.onEarth, Is.EqualTo(73.83m))
        Assert.That<decimal>(spaceAge.onMars, Is.EqualTo(39.25m))

    [<Test>]
    member tests.Age_on_jupiter() =
        let spaceAge = SpaceAge(901876382m)

        Assert.That<decimal>(spaceAge.onEarth, Is.EqualTo(28.58m))
        Assert.That<decimal>(spaceAge.onJupiter, Is.EqualTo(2.41m))

    [<Test>]
    member tests.Age_on_saturn() =
        let spaceAge = SpaceAge(3000000000m)

        Assert.That<decimal>(spaceAge.onEarth, Is.EqualTo(95.06m))
        Assert.That<decimal>(spaceAge.onSaturn, Is.EqualTo(3.23m))

    [<Test>]
    member tests.Age_on_uranus() =
        let spaceAge = SpaceAge(3210123456m)

        Assert.That<decimal>(spaceAge.onEarth, Is.EqualTo(101.72m))
        Assert.That<decimal>(spaceAge.onUranus, Is.EqualTo(1.21m))

    [<Test>]
    member tests.Age_on_neptune() =
        let spaceAge = SpaceAge(8210123456m)

        Assert.That<decimal>(spaceAge.onEarth, Is.EqualTo(260.16m))
        Assert.That<decimal>(spaceAge.onNeptune, Is.EqualTo(1.58m))

File: kotlin/space-age/src/test/kotlin/SpaceAgeTest.kt
import org.junit.Test
import org.junit.Ignore
import kotlin.test.assertEquals

class SpaceAgeTest {

    @Test
    fun ageOnEarth() {
        val age = SpaceAge(1000000000)

        assertEquals(31.69, age.onEarth())
    }

    @Test
    fun ageOnMercury() {
        val age = SpaceAge(2134835688)

        assertEquals(280.88, age.onMercury())
    }

    @Test
    fun ageOnVenus() {
        val age = SpaceAge(189839836)

        assertEquals(9.78, age.onVenus())
    }

    @Test
    fun ageOnMars() {
        val age = SpaceAge(2329871239L)

        assertEquals(39.25, age.onMars())
    }

    @Test
    fun ageOnJupiter() {
        val age = SpaceAge(901876382)

        assertEquals(2.41, age.onJupiter())
    }

    @Test
    fun ageOnSaturn() {
        val age = SpaceAge(3000000000L)

        assertEquals(3.23, age.onSaturn())
    }

    @Test
    fun ageOnUranus() {
        val age = SpaceAge(3210123456L)

        assertEquals(1.21, age.onUranus())
    }

    @Test
    fun ageOnNeptune() {
        val age = SpaceAge(8210123456L)

        assertEquals(1.58, age.onNeptune())
    }

}
